src/main/java/frc/team2412/robot/Subsystems/IndexerSubsystem.java

```diff
@@ -0,0 +1,304 @@
+package frc.team2412.robot.Subsystems;
+
+import com.revrobotics.CANSparkMax;
+
+import edu.wpi.first.wpilibj.DigitalInput;
+import edu.wpi.first.wpilibj.DoubleSolenoid;
+import edu.wpi.first.wpilibj.SpeedControllerGroup;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.team2412.robot.RobotMap;
+
+public class IndexerSubsystem extends SubsystemBase {
+	public CANSparkMax indexBackMotor;
+	public CANSparkMax indexFrontMotor;
+	public CANSparkMax indexMidMotor;
+	public SpeedControllerGroup indexMotors;
+	public SpeedControllerGroup indexSideMotors;
+
+	public DigitalInput back;
+	public DigitalInput backMid;
+	public DigitalInput mid;
+	public DigitalInput frontMid;
+	public DigitalInput front;
+	
+	public DoubleSolenoid frontClutch;
+	public DoubleSolenoid rearClutch;
+
+	public int numBalls = 0;
+	public boolean ballUnbalancedSide = false;
+
+	public IndexerSubsystem() {
+		// Motors
+		indexBackMotor = RobotMap.indexFrontMotor;
+		indexFrontMotor = RobotMap.indexBackMotor;
+		indexMidMotor = RobotMap.indexMidMotor;
+		indexMotors = new SpeedControllerGroup(indexFrontMotor, indexBackMotor, indexMidMotor);
+		indexSideMotors = new SpeedControllerGroup(indexFrontMotor, indexBackMotor);
+
+		// Sensors
+		back = RobotMap.BB;
+		backMid = RobotMap.BT;
+		mid = RobotMap.T;
+		frontMid = RobotMap.FT;
+		front = RobotMap.FF;
+		
+		//soulenoid
+		frontClutch = RobotMap.frontClutch;
+		rearClutch = RobotMap.rearClutch;
+	}
+
+	public void indexerLogic(boolean shoot) {
+		if (shoot) {
+			shoot();
+		} else {
+			switch (numBalls) {
+			case 0:
+				intake1();
+			case 1:
+				intake2();
+				switchBalls2(ballUnbalancedSide);
+			case 2:
+				intake3();
+				switchBalls34(ballUnbalancedSide);
+			case 3:
+				intake4();
+				switchBalls34(ballUnbalancedSide);
+			case 4:
+				intake5();
+			case 5:
+				System.out.println("im full");
+			}

```

Comment:
Consider using [SelectCommand](https://docs.wpilib.org/en/latest/docs/software/commandbased/convenience-features.html#selectcommand) for this, and using a [SequentialCommandGroup](https://docs.wpilib.org/en/latest/docs/software/commandbased/command-groups.html#sequentialcommandgroup) for the cases where multiple actions need to be done in sequence (note you can use `fooCommand.andThen(barCommand)` to easily create a sequential command group).